src/config_parse.c

```diff
@@ -135,7 +135,7 @@ static int parse_section_header_ext(git_config_parser *reader, const char *line,
 	}
 
 	*section_name = git_buf_detach(&buf);
-	return 0;
+	return base_name_len + 1 /* SP */ + 1 /* " */ + rpos + 1 /* " */ + 1 /* ] */;
```

Comment:
Can we really assume to have "static" length here? What about e.g. `[foo          "bar"]` or `[foo "bar"    ]`? Isn't that accepted by the parser? I don't think it currently is, but that leaves me wondering if we're stricter than git.git itself. A quick check shows that `[foo    "bar"]` is accepted, while `[foo "bar" ]` is not. I don't think we accept the former right now.

If we were to update our logic, would the above calculation still be correct? If possible, I'd instead prefer to do some pointer arithmetic between the start of the line and our end position.

Reply:
It is invalid to have a space between the closing quotes and the closing bracket. Both git and libgit2 refuse to parse such a file, so we should be fine.

I do agree that a more direct calculation would be nicer to have. The different sub-strings we use here made it a bit awkward so I did this to get the number. But again I do think the math will end up working out.